enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/CypherComparisonSupport.scala

```diff
@@ -145,44 +145,59 @@ trait CypherComparisonSupport extends CypherTestSupport {
                             params: Map[String, Any] = Map.empty): InternalExecutionResult = {
     // Never consider Morsel even if test requests it
     val expectSucceedEffective = expectSucceed - Configs.Morsel
-    val expectedDifferentResultsEffective = expectedDifferentResults - Configs.Morsel
 
-    val compareResults = expectSucceedEffective - expectedDifferentResultsEffective
-    val baseScenario =
-      if (expectSucceedEffective.scenarios.nonEmpty) extractBaseScenario(expectSucceedEffective, compareResults)
-      else TestScenario(Versions.Default, Planners.Default, Runtimes.Interpreted)
+    if (expectSucceedEffective.scenarios.nonEmpty) {
+      val expectedDifferentResultsEffective = expectedDifferentResults - Configs.Morsel
+      val compareResults = expectSucceedEffective - expectedDifferentResultsEffective
+      val baseScenario = extractBaseScenario(expectSucceedEffective, compareResults)
+      val explicitlyRequestedExperimentalScenarios = expectSucceedEffective.scenarios intersect Configs.Experimental.scenarios
 
-    val explicitlyRequestedExperimentalScenarios = expectSucceedEffective.scenarios intersect Configs.Experimental.scenarios
-    val positiveResults = ((Configs.AbsolutelyAll.scenarios ++ explicitlyRequestedExperimentalScenarios) - baseScenario).flatMap {
-      thisScenario =>
-        executeScenario(thisScenario, query, expectSucceedEffective.containsScenario(thisScenario), executeBefore, params, resultAssertionInTx)
-    }
+      val positiveResults = ((Configs.AbsolutelyAll.scenarios ++ explicitlyRequestedExperimentalScenarios) - baseScenario).flatMap {
+        thisScenario =>
+          executeScenario(thisScenario, query, expectSucceedEffective.containsScenario(thisScenario), executeBefore, params, resultAssertionInTx)
+      }
 
-    baseScenario.prepare()
-    executeBefore()
-    val baseResult = innerExecute(s"CYPHER ${baseScenario.preparserOptions} $query", params)
-    baseScenario.checkResultForSuccess(query, baseResult)
-    planComparisonStrategy.compare(expectSucceedEffective, baseScenario, baseResult)
+      //Must be run last and have no rollback to be able to do certain result assertions
+      val baseOption = executeScenario(baseScenario, query, expectedToSucceed = true, executeBefore, params, resultAssertionInTx = None, rollback = false)
```

Comment:
I am confused. Passing in `None` instead of `resultAssertionInTx` means, we are _not_ running inTx assertions for the baseScenario, doesn't it?

Reply:
Yes, I'm also confused about this part. 😆  My main intention with this work was not actually to solve the transaction thing, but to simplify the test configs. I then found some improvements that could be made in CypherComparisonSupport and me and Olivia thought I had fixed this card as a bonus. When I looked back at the code however I think you are right and that I actually have failed to do so. So I can either retry to solve it or you can regard this PR as a separete do-what-you-want work and not mark the transaction card as done.

Reply:
I think the code improvements are good to have, regardless of solving the inTx assertions. I'll put that back into prioritized TODO.